src/main/java/ch/njol/skript/effects/EffScriptFile.java

```diff
@@ -0,0 +1,129 @@
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.effects;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptCommand;
+import ch.njol.skript.config.Config;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.util.FileUtils;
+import ch.njol.util.Kleenean;
+
+@Name("Enable/Disable/Reload Script File")
+@Description("Enables, disables, or reloads a script file.")
+@Examples({"reload script \"test\"",
+			"enable script file \"testing\"",
+			"unload script file \"script.sk\""})
+@Since("INSERT VERSION")
+public class EffScriptFile extends Effect {
+	static {
+		Skript.registerEffect(EffScriptFile.class, "(1¦enable|1¦load|2¦reload|3¦disable|3¦unload) s(c|k)ript [file] %string%");
+	}
+	
+	private static final int ENABLE = 1, RELOAD = 2, DISABLE = 3;
+	
+	private int mark;
+	@Nullable
+	private Expression<String> fileName;
+	
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		mark = parseResult.mark;
+		fileName = (Expression<String>) exprs[0];
+		return true;
+	}
+	
+	@Override
+	public String toString(@Nullable Event e, boolean debug) {
+		return (mark == ENABLE ? "enable" : mark == RELOAD ? "disable" : mark == DISABLE ? "unload" : "") + " script file " + (fileName != null ? fileName.toString(e, debug) : "");
+	}
+	
+	@Override
+	protected void execute(Event e) {
+		String name = fileName != null ? fileName.getSingle(e) : "";
+		File f = SkriptCommand.getScriptFromName(name != null ? name : "");
+		if (f == null) {
+			Skript.error(name != null ? "Invalid script file name: " + name : "Invalid script file name!");
+			return;
+		}
+		switch (mark) {
+			case ENABLE: {
+				if (!f.getName().startsWith("-")) {
+					Skript.warning("Script file, " + name + ", is already enabled.");
+					return;
+				}
+				
+				try {
+					f = FileUtils.move(f, new File(f.getParentFile(), f.getName().substring(1)), false);
+				} catch (final IOException ex) {
+					Skript.error("Error while enabling script file: " + name);
+					return;
+				}
+				Config config = ScriptLoader.loadStructure(f);
+				ScriptLoader.loadScripts(config);
+				break;
+			}
+			case RELOAD: {
+				if (f.getName().startsWith("-")) {
+					Skript.warning("Script file, " + name + ", is disabled.");
+					return;
+				}
+				
+				if (!ScriptLoader.isAsync())
```

Comment:
Looks good. Have you checked how this interacts with async loading, though? It is not stable, but you can toggle it on with a hidden config option.